pandas/tests/series/test_timeseries.py

```diff
@@ -1012,3 +1012,22 @@ def test_get_level_values_box(self):
         index = MultiIndex(levels=levels, labels=labels)
 
         assert isinstance(index.get_level_values(0)[0], Timestamp)
+
+    def test_diff_sub_timedelta(self):
+        # GH 21980
+        arr = np.array([Timestamp('20130101 9:01'),
+                        Timestamp('20121230 9:02')])
+        exp = np.array([Timestamp('20121231 9:01'),
+                        Timestamp('20121229 9:02')])
+        res = arr - pd.Timedelta('1D')
+        tm.assert_numpy_array_equal(res, exp)
+
+    def test_diff_sub_timedelta_mixed(self):
```

Comment:
This PR probably also fixes addition right?  If so, pls include a test.  Possibly also for reversed ops?

Reply:
It should and I was thinking about reversed this morning actually so I'll include both today.